src/js/sentio/chart/chart_donut.js

```diff
@@ -0,0 +1,566 @@
+	 * creation and setup
+	 */
+	_instance.init = function(container){
+		// Create the DIV element
+		_element.div = container.append('div').attr('class', 'donut');
+
+		// set up the main donut svg
+		_svg = _element.div
+			.append('svg')
+			.attr('width', _width)
+			.attr('height', _height)
+			.append('g')
+			.attr('transform', 'translate(' + (_width / 2) +
+				',' + (_height / 2) + ')');
+
+		// set up the tooltip container
+		_tooltip = _element.div
+			.append('div')
+			.attr('class', 'tooltip');
+
+		_tooltip.append('div')
+			.attr('class', 'label');
+		_tooltip.append('div')
+			.attr('class', 'count');
+		_tooltip.append('div')
+			.attr('class', 'percent');
+
+		_instance.resize();
+
+		return _instance;
+	};
+
+	/*
+	 * Set the _instance data
+	 */
+	_instance.data = function(v) {
+		if(!arguments.length) { return _data; }
+		_data = v || [];
+		_data.forEach(function(d) {
+			d.enabled = true;
+		});
+		return _instance;
+	};
+
+	/*
+	 * Updates all the elements that depend on the size of the various components
+	 */
+	_instance.resize = function() {
+		return _instance;
+	};
+
+	/*
+	 * Redraw the graphic
+	 */
+	_instance.redraw = function() {
+		var radius = _radius;
+		var color = d3.scale.category10();
+
+		// Create the donut
+		var arc = d3.svg.arc()
+			.innerRadius(radius - _donutWidth)
+			.outerRadius(radius);
+
+		var pie = d3.layout.pie()
+			.value(function(d) { return d.value; })
+			.sort(null);
+
+		var g = _svg.selectAll(".arc")
+			.data(pie(_data));
+
+			g.transition(_duration)
+			.attrTween('d', function(d) {
+				var interpolate = d3.interpolate(this._current, d);
+				this._current = interpolate(0);
+				return function(t) {
+					return arc(interpolate(t));
+				};
+			});
+
+		var gEnter = g.enter();
+		gEnter.append("path")
+			.attr("class", "arc")
+			.each(function(d) { this._current = d; });
+			g.transition()
+			.duration(_duration)
+			.attrTween('d', function(d) {
+				if (this.tweened) return;
+				this.tweened = true;
+				// When this is the first draw, need a transition that
+				// ramps up like a gauge
+					var start = {
+						startAngle: 0,
+						endAngle: 0
+					};
+					var i = d3.interpolate(start, d);
+					return function(d1) { return arc(i(d1)); };
+			});
+
+		g.attr("class", "arc")
+			.attr("d", arc)
+			.attr('key', function(d) {
+				return d.data.key;
+			})
+			.attr('fill', function(d, i) {
+				return color(d.data.key);
+			})
+			.style('stroke', _arcStrokeColor);
+
+		var components = {
+			color: color,
+			pie: pie,
+			g: g,
+			arc: arc
+		};
+
+		if (_showTooltip || _highlightLegend) {
+			redrawTooltip(components);
+		}
+
+		if (_showLegend) {
+			redrawLegend(components);
+		}
+		return _instance;
+	};
+
+	/**
+	 * Private functions
+	 */
+	function redrawTooltip(components) {
+		var color = components.color;
+		var g = components.g;
+		var arc = components.arc;
+
+		// Mouse over a donut arc, show tooltip
+		g.on('mouseover', function(d) {
```

Comment:
You would only want to register this on handler on enter. Otherwise I dont know if its smart enough to not end up with multiple registered handlers